src/primval.rs

```diff
@@ -57,6 +59,34 @@ pub fn binary_op<'tcx>(bin_op: mir::BinOp, left: PrimVal, right: PrimVal) -> Eva
         })
     }
 
+    macro_rules! float_binops {
+        ($v:ident, $l:ident, $r:ident) => ({
+            match bin_op {
+                Add    => $v($l + $r),
+                Sub    => $v($l - $r),
+                Mul    => $v($l * $r),
+                Div    => $v($l / $r),
+                Rem    => $v($l % $r),
+
+                // invalid float ops
+                BitXor => unreachable!(),
+                BitAnd => unreachable!(),
+                BitOr  => unreachable!(),
+                Shl => unreachable!(),
+                Shr => unreachable!(),
+
+                // directly comparing floats is questionable
+                // miri could forbid it, or at least miri as rust const eval should forbid it
```

Comment:
I don't think it's so questionable that it should be forbidden. It's rarely the right thing to do, but in some special cases it might be correct.